File: python/leetcode005.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        l = 0
        ls = len(s)
        res = ""
        for i in range(ls):
            j = i + 1
            k = i - 1
            while j < ls and k >= 0:
                if s[j] == s[k]:
                    j = j + 1
                    k = k - 1
                else:
                    break
            if j-k-1 > l:
                l = j-k-1
                res = s[k+1:j]
            if i + 1 < ls and s[i] == s[i+1]:
                j = i + 2
                k = i - 1
                while j < ls and k >= 0:
                    if s[j] == s[k]:
                        j = j + 1
                        k = k - 1
                    else:
                        break
                if j - k - 1 > l:
                    l = j - k - 1
                    res = s[k + 1:j]
        return res

if __name__ == "__main__":
    s = Solution()
    print(s.longestPalindrome('ccc'))

File: leetcodec++/leetcode005.cpp
//
// Created by 黄敏辉 on 2019-06-10.
//

class Solution {
public:
    string longestPalindrome(string s) {
        int len = 0, start = 0;
        for (int i = 0; i < s.size(); i++){
            int j = 0;
            while (i - j >= 0 && i + j < s.size() && s[i-j] == s[i+j]){
                if (2 * j + 1 > len){
                    len = 2*j+1;
                    start = i - j;
                }
                j++;
            }
            j = 0;
            while (i - j >= 0 && i + j + 1 < s.size() && s[i-j] == s[i+j+1]){
                if (2 * j + 2 > len){
                    len = 2*j+2;
                    start = i - j;
                }
                j++;
            }
        }
        return s.substr(start, len);
    }
};